extension/data.ts

```diff
@@ -199,190 +188,152 @@ class RcxDict {
     // i = 1: skip header
     for (let i = 1; i < buffer.length; ++i) {
       const ruleOrReason = buffer[i].split('\t');
-
       if (ruleOrReason.length === 1) {
         this.difReasons.push(ruleOrReason[0]);
       } else if (ruleOrReason.length === 4) {
-        const o: DeinflectionRule = {
+        const deinflectionRule: DeinflectionRule = {
           from: ruleOrReason[0],
           to: ruleOrReason[1],
           typeMask: parseInt(ruleOrReason[2]),
           reasonIndex: parseInt(ruleOrReason[3]),
         };
-
-        if (currentLength !== o.from.length) {
-          currentLength = o.from.length;
+        if (currentLength !== deinflectionRule.from.length) {
+          currentLength = deinflectionRule.from.length;
           group = { fromLength: currentLength, rules: [] };
           this.difRules.push(group);
         }
-        group.rules.push(o);
+        group.rules.push(deinflectionRule);
       }
     }
   }
 
   find(data: string, text: string): string | null {
-    const tlen = text.length;
-    let beg = 0;
+    const textLength = text.length;
+    let beginning = 0;
     let end = data.length - 1;
-    let i;
-    let mi;
-    let mis;
-
-    while (beg < end) {
-      mi = (beg + end) >> 1;
-      i = data.lastIndexOf('\n', mi) + 1;
-
-      mis = data.substr(i, tlen);
+    while (beginning < end) {
+      const middle = (beginning + end) / 2;
+      const i = data.lastIndexOf('\n', middle) + 1;
+      const mis = data.substr(i, textLength);
       if (text < mis) {
         end = i - 1;
       } else if (text > mis) {
-        beg = data.indexOf('\n', mi + 1) + 1;
+        beginning = data.indexOf('\n', middle + 1) + 1;
       } else {
-        return data.substring(i, data.indexOf('\n', mi + 1));
+        return data.substring(i, data.indexOf('\n', middle + 1));
       }
     }
     return null;
   }
 
-  deinflect(word: string) {
-    const r = [];
-    const have: { [key: string]: number } = {};
-    let o;
-
-    o = { word: word, type: 0xff, reason: '' } as Deinflection;
-    r.push(o);
+  deinflect(word: string): Deinflection[] {
+    const possibleDeinflections: Deinflection[] = [
+      { word, type: 0xff, reason: '' },
+    ];
+    const have: Record<string, number> = {};
     have[word] = 0;
 
-    let i;
-    let j;
-    let k;
-
-    i = 0;
-    do {
-      word = r[i].word;
-      const wordLen = word.length;
-      const type = r[i].type;
+    for (let i = 0; i < possibleDeinflections.length; i++) {
+      const currentDeinflection = possibleDeinflections[i].word;
```

Comment:
it seems odd to call this `currentDeinflection` since it's a sub field of `possibleDeinflections`

Either maybe the `word` subfield should be updated to be a better name for what it represents or we should use something like `currentDeinflectionWord` here or simlar.

Reply:
Good catch again. This was an overlook on my part. It very well should be that. 

I'm thinking `curDeinflectionWord` to cut down a tad bit on the length. Starting to look more and more like Java.

Sound good?